packages/bcer-api/app/src/sales/sales.controller.ts

```diff
@@ -62,6 +78,112 @@ export class SalesReportController {
     @Request() req: RequestWithUser,
   ): Promise<SalesReportRO[]> {
     const sales = await this.service.getSalesReports();
-    return sales.map((salesReport: SalesReportEntity) => salesReport.toResponseObject());
+    return sales.map((salesReport: SalesReportEntity) =>
+      salesReport.toResponseObject(),
+    );
+  }
+
+  @ApiOperation({ summary: 'Create Sales Reports from Submission' })
+  @ApiResponse({ status: HttpStatus.CREATED, type: [SalesReportRO] })
+  @HttpCode(HttpStatus.CREATED)
+  @Roles('user')
+  @UseGuards(BusinessGuard)
+  @Post('/:submissionId')
+  async createProductSoldSalesReports(
+    @Param('submissionId') submissionId: string,
+    @Request() req: RequestWithUser,
+  ): Promise<any> {
+    const submission = await this.submissionService.getOne(submissionId);
+
+    if (submission.businessId !== req.ctx.businessId) {
+      throw new ForbiddenException(
+        'This user does not have access to this submission',
+      );
+    }
+
+    const saleReportsData = submission.data.saleReports;
+    const locationId = submission.data.locationId;
+    const year = submission.data.year;
+    const businessId = submission.businessId;
+
+    // validation
+    const violations = saleReportsData.filter(
+      ({
+        containerCapacity,
+        cartridgeCapacity,
+        containers,
+        cartridges,
+      }: any) => {
+        return (
+          (!containerCapacity && !cartridges && !containers && !cartridges) ||
+          (containerCapacity && !cartridgeCapacity && !containers && cartridges) ||
+          (!containerCapacity && cartridgeCapacity && containers && !cartridges) ||
+          (!containerCapacity && !cartridgeCapacity && !containers && cartridges) ||
+          (!containerCapacity && !cartridgeCapacity && containers && !cartridges) ||
+          (!containerCapacity && !cartridgeCapacity && containers && cartridges) ||
+          (containerCapacity && cartridgeCapacity && !containers && !cartridges)
+        );
+      },
+    );
+
+    if(violations && violations.length > 0) {
+      throw new Error('1.	“At least one value for Cartridges or Containers sold as well as capacity must be provided for a given product. The capacity column and the number sold column should match. Both should either be for containers or for cartridges.”')
+    }
+
+    // replace existing sales reports, deleting the existing one.
+    if (submission.data.isSubmitted) {
+      const existingProductSolds = await this.service.remove(locationId, year);
```

Comment:
We've confirmed that we don't need these to be archived instead?

Reply:
Yes, did confirm with Killian. no need the archived one. If user edit the reports, just delete the old one and replace with the new one. This is also showing the warning message on frontend.

![image](https://user-images.githubusercontent.com/46575401/134403396-19ada00e-d77c-4380-b137-a5921e8e5cc8.png)